File: src/ylva/ynab/model/subtransaction.py
from dataclasses import dataclass, field

from mashumaro import DataClassDictMixin, field_options

from . import Id


@dataclass
class Subtransaction(DataClassDictMixin):
    id_: Id = field(metadata=field_options(alias="id"))
    transaction_id: Id
    amount: float
    memo: str
    payee_id: Id
    payee_name: str
    category_id: Id
    category_name: str
    transfer_account_id: Id
    transfer_transaction_id: Id
    deleted: bool

File: src/model/ynab/subtransaction.rs
use crate::model::ynab::id::Id;

#[derive(Debug, serde::Serialize, serde::Deserialize)]
pub struct Subtransaction {
    pub id: Id,
    pub transaction_id: Id,
    pub amount: i64,
    pub memo: String,
    pub payee_id: Id,
    pub payee_name: String,
    pub category_id: Id,
    pub category_name: String,
    pub transfer_account_id: Id,
    pub transfer_transaction_id: Id,
    pub deleted: bool,
}